controller/test/alarm/alarm_test.cpp

```diff
@@ -20,10 +20,45 @@ TEST_F(AlarmTest, AddOneAlarm) {
   ASSERT_TRUE(alarm_available());
 }
 
+TEST_F(AlarmTest, RemoveOneAlarm) {
+  alarm_remove();
```

Comment:
Is the intent of this test to check that we can remove an alarm after having added one alarm (in `alarm_add`)?  Or is the intent to check that alarm_remove() is OK to call if there are no alarms present?

Right now it's doing the former, but that's relying on tests being run in order, and on nobody modifying AddOneAlarm.  To do the latter, we'd need to reolve the TODO in `SetUp`.

Reply:
The former; I was relying on test order (I wasn't sure what best practice was w.r.t. order-independence; took a cue from NotAvailableAfterInit.) If we want tests to be order-independent I'll add an alarm before removing one.

Reply:
> If we want tests to be order-independent I'll add an alarm before removing one.

Yes, I think we want tests to be order-independent, otherwise modifying one test can break another.

To really make them hermetic we should reset the alarm state before each test in `SetUp`.

Reply:
Okay! I've made the add and the remove test a single case so that they're order-independent. 

I'm not sure what you mean by "reset the alarm state before each test in SetUp". Do you mean for each test suite (by suite, I mean the collection of tests in a single file that are testing the same module) we should include an alarm reset in that suite's `SetUp`?

Reply:
Ah, I think I understand.

A nonintuitive thing about gunit is that the test suite's `SetUp()` function is called before every test.  If you want to do one-time setup that's not repeated for every test, the function is `SetUpTestSuite()`.

> Do you mean for each test suite (by suite, I mean the collection of tests in a single file that are testing the same module) we should include an alarm reset in that suite's SetUp?

For instance, right now, `OverflowAlarmQueue ` leaves the alarm module with many alarms in its stack.  So then that affects the next test, which has to clear out the alarms.  It would be nice if we'd clear the alarm module's state before each test so that one test *can't* affect the next one.

It may be as simple as moving this code into `SetUp`:

```
  while (alarm_available()) {
    alarm_remove();
  }
```

Reply:
oh! I see! I didn't realize setup was called before every test case. Thank you for the explanation. I've made the suggested change.